timemachine/cpp/src/device_buffer.cu

```diff
@@ -6,20 +6,24 @@
 
 namespace timemachine {
 
-template <typename T> T *allocate(const std::size_t length) {
+namespace {
+
+template <typename T> T *allocate_gpu_memory(const std::size_t length) {
```

Comment:
try:

`template <typename T> static T *allocate(const std::size_t length)`

Reply:
(`static` limits accessibility to the translation unit/file.)

Reply:
Whoops statics works as well, had borked the syntax yesterday though. Will change

Reply:
Referred to static in https://github.com/proteneer/timemachine/pull/1402/commits/8f2da9315309689fafe79942e6aa43affdd9af2f

Though kept one unnamed namespace for a struct definition that is local to the file.